src/librustc/diagnostics.rs

```diff
@@ -1661,6 +1661,35 @@ To understand better how closures work in Rust, read:
 https://doc.rust-lang.org/book/closures.html
 "##,
 
+E0579: r##"
+The `main` function was incorrectly declared.
+
+Erroneous code example:
+
+```compile_fail,E0579
+fn main() -> i32 { // error: main function has wrong type
+    0
+}
+```
+
+The `main` function prototype should always be the following:
```

Comment:
Rust doesn't have function prototypes so this term is sort of out of the blue. I think showing the correct shape of the function as in the next code example is fine.